python/lsst/afw/display/saturated.cc

```diff
@@ -0,0 +1,176 @@
+                       ) {
+            *loc = _value;
+        }
+    private:
+        float _value;
+    };
+}
+
+template<typename ImageT>
+void
+replaceSaturatedPixels(ImageT & rim,    // R image (e.g. i)
+                       ImageT & gim,    // G image (e.g. r)
+                       ImageT & bim,    // B image (e.g. g)
+                       int borderWidth,	// width of border used to estimate colour of saturated regions
+                       float saturatedPixelValue // the brightness of a saturated pixel, once fixed
+                      )
+{
+    int const width = rim.getWidth(), height = rim.getHeight();
+    int const x0 = rim.getX0(), y0 = rim.getY0();
+
+    if (width != gim.getWidth() || height != gim.getHeight() || x0 != gim.getX0() || y0 != gim.getY0()) {
+        throw LSST_EXCEPT(lsst::pex::exceptions::InvalidParameterError,
+                          str(boost::format("R image has different size/origin from G image "
+                                            "(%dx%d+%d+%d v. %dx%d+%d+%d")
+                              % width % height % x0 % y0
+                              % gim.getWidth() % gim.getHeight() % gim.getX0() % gim.getY0()));
+
+    }
+    if (width != bim.getWidth() || height != bim.getHeight() || x0 != bim.getX0() || y0 != bim.getY0()) {
+        throw LSST_EXCEPT(lsst::pex::exceptions::InvalidParameterError,
+                          str(boost::format("R image has different size/origin from B image "
+                                            "(%dx%d+%d+%d v. %dx%d+%d+%d")
+                              % width % height % x0 % y0
+                              % bim.getWidth() % bim.getHeight() % bim.getX0() % bim.getY0()));
+
+    }
+
+    bool const useMaxPixel = !lsst::utils::isfinite(saturatedPixelValue);
+
+    SetPixels<typename ImageT::Image>
+        setR(*rim.getImage()),
+        setG(*gim.getImage()),
+        setB(*bim.getImage()); // functors used to set pixel values
+
+    // Find all the saturated pixels in any of the three image
+    int const npixMin = 1;              // minimum number of pixels in an object
+    int const SAT = rim.getMask()->getPlaneBitMask("SAT");
+    detection::Threshold const satThresh(SAT, detection::Threshold::BITMASK);
+
+    detection::FootprintSet       sat(*rim.getMask(), satThresh, npixMin);
+    sat.merge(detection::FootprintSet(*gim.getMask(), satThresh, npixMin));
+    sat.merge(detection::FootprintSet(*bim.getMask(), satThresh, npixMin));
+    // go through the list of saturated regions, determining the mean colour of the surrounding pixels
+    typedef detection::FootprintSet::FootprintList FootprintList;
+    PTR(FootprintList) feet = sat.getFootprints();
+    for (FootprintList::iterator ptr = feet->begin(), end = feet->end(); ptr != end; ++ptr) {
+        PTR(detection::Footprint) foot = *ptr;
+        PTR(detection::Footprint) bigFoot = growFootprint(*foot, borderWidth);
```

Comment:
I'm surprised `growFootprint` doesn't have a namespace (`detection::`) prepended, yet it compiles.  What am I missing?